integration/query_frontend_test.go

```diff
@@ -89,21 +89,47 @@ func TestQueryFrontendTLSWithBlocksStorageViaFlags(t *testing.T) {
 		minio := e2edb.NewMinio(9000, BlocksStorageFlags["-experimental.blocks-storage.s3.bucket-name"])
 		require.NoError(t, s.StartAndWaitReady(minio))
 
-		// setup tls
-		cmd := exec.Command("bash", "certs/genCerts.sh", "certs", "1")
-		require.NoError(t, cmd.Run())
-		require.NoError(t, copyFileToSharedDir(s, integrationHomeFolder+clientCertFile, clientCertFile))
-		require.NoError(t, copyFileToSharedDir(s, integrationHomeFolder+clientKeyFile, clientKeyFile))
-		require.NoError(t, copyFileToSharedDir(s, integrationHomeFolder+caCertFile, caCertFile))
-		require.NoError(t, copyFileToSharedDir(s, integrationHomeFolder+serverCertFile, serverCertFile))
-		require.NoError(t, copyFileToSharedDir(s, integrationHomeFolder+serverKeyFile, serverKeyFile))
+		// set the ca
+		ca := ca.New("Cortex Test")
+
+		// Ensure the entire path of directories exist.
+		require.NoError(t, os.MkdirAll(filepath.Join(s.SharedDir(), "certs"), os.ModePerm))
+
+		require.NoError(t, ca.WriteCACertificate(
+			filepath.Join(s.SharedDir(), caCertFile),
+		))
+
+		// server certificate
+		require.NoError(t, ca.WriteCertificate(
+			&x509.Certificate{
+				Subject: pkix.Name{
+					CommonName: "client",
+				},
+				ExtKeyUsage: []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth},
+			},
+			filepath.Join(s.SharedDir(), clientCertFile),
+			filepath.Join(s.SharedDir(), clientKeyFile),
+		))
+		require.NoError(t, ca.WriteCertificate(
+			&x509.Certificate{
+				Subject: pkix.Name{
+					CommonName: "server",
+				},
+				DNSNames: []string{
+					"querier.frontend-client",
+					"ingester.client",
+				},
+				ExtKeyUsage: []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth},
+			},
+			filepath.Join(s.SharedDir(), serverCertFile),
+			filepath.Join(s.SharedDir(), serverKeyFile),
+		))
```

Comment:
Just a reformatting to make it a bit more readable (for me :)). Feel free to ignore if you don't like it.

```suggestion
		require.NoError(t, ca.WriteCACertificate(filepath.Join(s.SharedDir(), caCertFile)))

		// server certificate
		require.NoError(t, ca.WriteCertificate(
			&x509.Certificate{Subject: pkix.Name{CommonName: "client"}, ExtKeyUsage: []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth}},
			filepath.Join(s.SharedDir(), clientCertFile),
			filepath.Join(s.SharedDir(), clientKeyFile),
		))
		require.NoError(t, ca.WriteCertificate(
			&x509.Certificate{
				Subject:     pkix.Name{CommonName: "server"},
				DNSNames:    []string{"querier.frontend-client", "ingester.client"},
				ExtKeyUsage: []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth},
			},
			filepath.Join(s.SharedDir(), serverCertFile),
			filepath.Join(s.SharedDir(), serverKeyFile),
		))
```

Reply:
I have collapsed a few of the substructs :slightly_smiling_face: